### ISSUE TITLE
Running dependencyUpdates task in combination with publishing build scan can fail publishing build scans

### ISSUE THREAD

**Issue by breskeby** [ben-manes/gradle-versions-plugin#315]
This is not strictly speaking a bug in the `gradle-versions-plugin` but it makes using the plugin harder for some teams (This issue was reported by the Junit5 team originally)

Applying the plugin on all projects in a multi project build and running `./gradlew dependencyUpdates --parallel --scan` can result in an error in generating a build scan. I looked into this today and noticed the following cause for this:

1. the plugin seems to be designed by applying it on the root project only as it iterates over all projects via `projects.allProjects`

2. the plugin creates a copy of each configuration which results in configurations named like `:my-project:myConfigurationCopy`

3. This means by applying the plugin on each subproject  the configuration `:my-project:myConfigurationCopy` will be created multiple times (e.g. in the `: dependencyUpdates` task _AND_ in `:my-project:dependencyUpdates` task)

4. The scan plugin doesn't deal well with configurations of the same exact path.

 In general of course there are multiple ways to solve this.

1. We probably will deprecate the option to create two configurations like that with the same name and path in gradle at one point (will discuss this with the gradle team)
2. Fix the scan plugin to deal with these kind of duplicates.
3. You could probably provide a warning when the gradle versions plugin is applied on a non root project
4. The plugin could move to use detachedConfigurations for inspecting configurations. 

1. and 2 can probably done at one point by the gradle team. But as a heads up I think to provide a better user experience the plugin can be improved by applying option 3 and 4.

What do you think? I'm happy to support you if you need further help on this.

**Comment by ben-manes** [comment]
Hi Rene,

Your description of how the plugin works is correct. The process hasn't changed much from its [original](https://discuss.gradle.org/t/add-report-to-display-dependency-updates/540) version, circa Gradle 1.x.

1. The responsibility for naming the configuration is Gradle's, as it is done via `copy()`. This plugin does not depend on the naming chosen.
3. Surprisingly, this is what some users wanted and prefer. We'd have to dig into the issue tracker as applying to the root is how I always used it and expectation. I suspect this was due to how team responsibilities across projects were divided.
4. This was how the original version worked, but we ran into bugs within Gradle and found `Configuration#copy()` was less risky. In particular, I recall that `RepositoryHandler` was a source of problems due to its incorrect duplicate detection.

In #263, the Gradle team indicated that accessing subprojects should be replaced by a per-project task and a root-level merge task. A major refactoring should probably take that into account.

Unfortunately, I do not have the time for this project and only promise maintenance for compatibility.  The community has been great with fixes and improvements, but there is no one driving this plugin anymore. I have always viewed it as a stop-gap until Gradle adds the feature into its core and feel it is very much a feature to directly support.

PRs are welcome and I am happy to offer commit access to those interested.
